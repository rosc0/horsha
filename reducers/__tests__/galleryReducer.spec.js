import reducer, { initialState as state } from '../gallery'
import types from '../../constants/actionTypes'

import image_input from '../../../test/api/inputs/image'

const expectResponses = {
  add: {
    images: [image_input.POST],
  },
  clear: {
    images: [],
  },
}

describe('Settings Reducer', () => {
  it('should return initial state', () => {
    expect(reducer()).toBe(state)
    expect(reducer()).toMatchSnapshot()
  })

  it('should GALLERY_ADD_IMAGES', () => {
    const act = reducer(state, {
      type: types.GALLERY_ADD_IMAGES,
      images: [image_input.POST],
    })

    expect(act).toEqual(expectResponses.add)
    expect(act).toMatchSnapshot()
  })

  it('should GALLERY_CLEAR_IMAGES', () => {
    const act = reducer(state, {
      type: types.GALLERY_CLEAR_IMAGES,
    })

    expect(act).toEqual(expectResponses.clear)
    expect(act).toMatchSnapshot()
  })
})
