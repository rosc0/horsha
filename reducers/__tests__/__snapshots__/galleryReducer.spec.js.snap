// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Reducer should GALLERY_ADD_IMAGES 1`] = `
Object {
  "images": Array [
    Object {
      "cropped": true,
      "filename": "IMG_0111.HEIC",
      "height": 3024,
      "isStored": true,
      "playableDuration": 0,
      "uri": "assets-library://asset/asset.HEIC?id=CC95F08C-88C3-4012-9D6D-64A413D254B3&ext=HEIC",
      "width": 4032,
    },
  ],
}
`;

exports[`Settings Reducer should GALLERY_CLEAR_IMAGES 1`] = `
Object {
  "images": Array [],
}
`;

exports[`Settings Reducer should return initial state 1`] = `
Object {
  "images": Array [],
}
`;
